File: transcription-server/src/models/transcript.py
from pydantic import BaseModel


class TranscriptSegment(BaseModel):
    start: float
    end: float
    text: str


class TranscriptResult(BaseModel):
    text: str
    segments: list[TranscriptSegment]
    language: str
    duration: float


class TranscribeURLRequest(BaseModel):
    file_path: str
    language: str | None = None

File: Trolly/Domain/Models/Transcript.swift
import Foundation

struct Transcript: Codable, Equatable, Sendable {
    let text: String
    let segments: [TranscriptSegment]
    let language: String
    let duration: TimeInterval
}

struct TranscriptSegment: Codable, Equatable, Sendable {
    let start: TimeInterval
    let end: TimeInterval
    let text: String
}
